### DISCUSSION TITLE
Licensing issue with ffmpeg

### DISCUSSION THREAD

**Discussion by ozancaglayan** [openai/whisper#798] [Q&A] [answered]
Hello,

I see that whisper package is licensed with MIT. However, it uses `ffmpeg` under the hood, although it doesn't list it as a dependency (as its installation is system-dependent). Now `ffmpeg` is a complex package, some of it being licensed under GPL, other parts under LGPL. Does this invalidate the license of `whisper`? Is there an obstacle in replacing the use of ffmpeg with
`torchaudio.load` or `librosa`, or some other simpler alternatives?

Thanks!

**Comment by glangford** [comment, marked as answer]
License contamination is a complicated discussion, but here is one line of thinking.

* I don't know that `torchaudio` successfully evades (L)GPL since it uses `ffmpeg` as well; see 
https://github.com/pytorch/audio/releases
and search for `ffmpeg`.

* `whisper` in its current form uses `ffmpeg` here via `ffmpeg-python`
https://github.com/openai/whisper/blob/9f70a352f9f8630ab3aa0d06af5cb9532bd8c21d/whisper/audio.py#L22
In this code is the comment
`# This launches a subprocess to decode audio while down-mixing and resampling as necessary.`

As far as I know, the general consensus seems to be that this type of subprocess invocation preserves separation of components and avoids license contamination. See for example the discussion here:
https://www.ifross.org/?q=en/program-forks-gpl-licensed-program-system-or-vice-versa-call-derivative-work

> Starting an application from another application and exchanging data between the calling and the called program via command line, files or pipes is a normal and established procedure. This happens, for example, nearly every time a program is started from the shell. If this procedure implied that the called program is assumed a derivative work of the calling program and the shell in use is licensed under the GPL such as the standard shell bash, it would no longer be possible to run a non-GPL application from shell level. This would not reflect the intention of the GPL and would contradict common practice accepted from the authors of the GPL over many years.
